src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1266,11 +1266,12 @@ private void SpillEvalStack()
             for (int i = 0; i < _evalStack.Count; i++)
             {
                 IOperation operation = _evalStack[i];
-                if (operation.Kind != OperationKind.FlowCaptureReference)
+                // Declarations cannot have control flow, so we don't need to spill them.
+                if (operation.Kind != OperationKind.FlowCaptureReference && operation.Kind != OperationKind.DeclarationExpression)
                 {
                     int captureId = _availableCaptureId++;
 
-                    AddStatement(new FlowCapture(captureId, operation.Syntax, Visit(operation))
```

Comment:
I'm surprised by the change here. We don't need to visit `operation`?

Reply:
Everything that is on `_evalStack` has already been visited. This extra call to Visit is actually a bug that was causing issues with DeconstructionFlow_07.

---
In reply to: [181832673](https://github.com/dotnet/roslyn/pull/26108#discussion_r181832673) [](ancestors = 181832673)